rpcs3/Emu/Cell/lv2/sys_rsx.cpp

```diff
@@ -194,10 +195,13 @@ error_code sys_rsx_context_iomap(u32 context_id, u32 io, u32 ea, u32 size, u64 f
 
 	io >>= 20, ea >>= 20, size >>= 20;
 
+	std::scoped_lock lock(s_rsxmem_mtx);
+
 	for (u32 i = 0; i < size; i++)
 	{
+		const u32 prev_ea = std::exchange(RSXIOMem.ea[io + i].raw(), ea + i);
+		if (prev_ea < 0xC00) RSXIOMem.io[prev_ea].raw() = 0xFFFF; // Clear previous mapping if exists
```

Comment:
I hope this won't race in the same set where an entry can undo another one.

Reply:
The only place which reads from RSXIOMem.io is gcm unmap events, which are under vm::writer_lock, rsxiomsp is under vm::resder_lock so it is not an issue.